File: app/src/main/java/com/messi/languagehelper/viewmodels/TranDictViewModel.java
package com.messi.languagehelper.viewmodels;

import android.app.Application;

import androidx.annotation.NonNull;
import androidx.lifecycle.AndroidViewModel;
import androidx.lifecycle.LiveData;
import androidx.lifecycle.MutableLiveData;

import com.messi.languagehelper.bean.RespoData;
import com.messi.languagehelper.box.Dictionary;
import com.messi.languagehelper.box.Record;
import com.messi.languagehelper.repositories.TranDictRepository;

public class TranDictViewModel extends AndroidViewModel {

    private MutableLiveData<RespoData<Record>> mTranRespoData;
    private MutableLiveData<RespoData<Dictionary>> mDictRespoData;
    private MutableLiveData<Boolean> isRefreshTran;
    private MutableLiveData<Boolean> isRefreshDict;
    private TranDictRepository mRepository;

    public TranDictViewModel(@NonNull Application application) {
        super(application);
        mRepository = new TranDictRepository(getApplication());
        mTranRespoData = mRepository.mRespoData;
        isRefreshTran = mRepository.isRefreshTran;
        isRefreshDict = mRepository.isRefreshDict;
        mDictRespoData = mRepository.mDictRespoData;
    }

    public void initSample() {
        mRepository.initSample();
    }

    public void loadTranData(boolean isRefresh){
        mRepository.loadTranData(isRefresh);
    }

    public void tranDict(){
        mRepository.tranDict();
    }

    public void loadDictData(boolean isRefresh){
        mRepository.loadDictData(isRefresh);
    }

    public void getDict(){
        mRepository.getDict();
    }

    public LiveData<Boolean> isRefreshTran(){
        return isRefreshTran;
    }

    public LiveData<RespoData<Record>> getTranRespoData(){
        return mTranRespoData;
    }

    public LiveData<Boolean> isRefreshDict(){
        return isRefreshDict;
    }

    public LiveData<RespoData<Dictionary>> getDictRespoData(){
        return mDictRespoData;
    }

    public TranDictRepository getRepository(){
        return mRepository;
    }

}

File: app/src/main/java/com/messi/languagehelper/viewmodels/TranDictViewModel.kt
package com.messi.languagehelper.viewmodels

import android.app.Application
import androidx.lifecycle.AndroidViewModel
import androidx.lifecycle.LiveData
import androidx.lifecycle.MutableLiveData
import androidx.lifecycle.viewModelScope
import com.messi.languagehelper.bean.RespoData
import com.messi.languagehelper.box.Dictionary
import com.messi.languagehelper.box.Record
import com.messi.languagehelper.repositories.TranDictRepository
import kotlinx.coroutines.Dispatchers
import kotlinx.coroutines.launch

class TranDictViewModel(application: Application) : AndroidViewModel(application) {

    private val mTranRespoData: MutableLiveData<RespoData<Record>>
    private val mDictRespoData: MutableLiveData<RespoData<Dictionary>>
    private val isRefreshTran: MutableLiveData<Boolean>
    private val isRefreshDict: MutableLiveData<Boolean>
    var repository: TranDictRepository = TranDictRepository(getApplication())

    init {
        mTranRespoData = repository.mRespoData
        isRefreshTran = repository.isRefreshTran
        isRefreshDict = repository.isRefreshDict
        mDictRespoData = repository.mDictRespoData
    }

    fun initSample() {
        viewModelScope.launch(Dispatchers.IO) {
            repository.initSample()
        }
    }

    fun loadTranData(isRefresh: Boolean) {
        viewModelScope.launch(Dispatchers.IO) {
            repository.loadTranData(isRefresh)
        }
    }

    fun tranDict() {
        viewModelScope.launch(Dispatchers.IO) {
            repository.tranDict()
        }
    }

    fun loadDictData(isRefresh: Boolean) {
        viewModelScope.launch(Dispatchers.IO) {
            repository.loadDictData(isRefresh)
        }
    }

    fun getDict() {
        viewModelScope.launch() {
            repository.getDict()
        }
    }

    fun isRefreshTran(): LiveData<Boolean> {
        return isRefreshTran
    }

    val tranRespoData: LiveData<RespoData<Record>>
        get() = mTranRespoData

    fun isRefreshDict(): LiveData<Boolean> {
        return isRefreshDict
    }

    val dictRespoData: LiveData<RespoData<Dictionary>>
        get() = mDictRespoData
}